packages/core/chrome/core-chrome-browser-internal/src/ui/header/nav_link.tsx

```diff
@@ -171,6 +189,14 @@ export function createRecentNavLink(
     onClick(event: React.MouseEvent<HTMLButtonElement, MouseEvent>) {
       if (event.button === 0 && !isModifiedOrPrevented(event)) {
         event.preventDefault();
+        <EuiScreenReaderLive focusRegionOnTextChange>
+          {i18n.translate('core.ui.recentLinks.linkItem.screenReaderAnnouncement.title', {
+            defaultMessage: 'type: {title}',
+            values: {
+              title: navLink?.title,
+            },
+          })}
+        </EuiScreenReaderLive>;
```

Comment:
I don't think this usage of `<EuiScreenReaderLive>` within an onClick event is correct - it's not clear that it's actually rendering anywhere meaningful.

Instead of making the screen reader announcement imperative (e.g. firing on click), I think you need to make it reactive (e.g. responding to `document.title` changes, which it then stores in state), and only located in 1 place across all apps. Does that make sense? Happy to chat more if not.

Reply:
I'll plus one to @cee-chen about responding to `document.title` and having this centrally located. I've experienced NVDA announce changes inconsistently when the live region is added late. Having it in the DOM on first `render()` and then update with `document.title` should give a consistent experience.

Reply:
Ok thank you both! I think I'm following - updated in [226e5e0](https://github.com/elastic/kibana/pull/150461/commits/226e5e0e291c7fa8744b652fbe1dd823dfbe95ac) :

Reply:
https://user-images.githubusercontent.com/20343860/218811941-0c8da481-d367-412e-a766-31b670446c33.mp4